### ISSUE TITLE
KernelShap class problem (binary)

### ISSUE THREAD

**Issue by Atnamass** [shap/shap#1652]
Hello,

I have been using ShapKernel explainer in order to generate explanations for my own image data set.
I have also implemented another model (CNN) and included it into the code instead of the VGG16 one.

In the case VGG16  the image net dataset is used (999 classes) and everytime explanation is generated the results show the prediction for the top 3 classes.
In my case I only have 2 classes and when the results are displayed the prediction is shown only for the top class. What I would like is to have the prediction for both classes shown - so that it is easier to understand the contribution to each of the class on the image/ or why there is a higher probability of that class for each image.
The same as it is shown in the ImageNet example: https://slundberg.github.io/shap/notebooks/ImageNet%20VGG16%20Model%20with%20Keras.html

**Current SHAP code:**
I have been trying to make changes by doing some modifications to this last part:

def f(z):
       return model.predict(preprocess_input(mask_image(z, segments_slic, images_bleeding_val_sample, 255)))

For example:

def f(z):
       return model.predict_proba(preprocess_input(mask_image(z, segments_slic, images_bleeding_val_sample, 255)))

Has anybody run in a similar issue using their own model?
The other option I thought of doing was to make changes in the CNN model code already - and train it again, in order for it to produce two outputs for each class.

**Example of the explanation:**

![pastedImage1](https://user-images.githubusercontent.com/49812933/102109341-9b560d80-3e3c-11eb-941a-59e75b0671d2.png)

![pastedImage2](https://user-images.githubusercontent.com/49812933/102109369-a14bee80-3e3c-11eb-8274-a1f997afc17a.png)

What I would like to generate (but with the two classes I have):

![pastedImage](https://user-images.githubusercontent.com/49812933/102109388-a7da6600-3e3c-11eb-9fcc-55e0adc1461e.png)

CNN model, training part of the code:

# Training

Now, we save the model and save it to disk. For this, we use this Keras code as the blue print: https://gist.github.com/fchollet/0830affa1f7f19fd47b06d4cf89ed44d.

[4]:
#dimensions of our images.
img_width, img_height = 150, 150
​
nb_train_samples = training_data_length
nb_validation_samples = validation_data_length
epochs = 50
batch_size = 16
​
if K.image_data_format() == 'channels_first':
    input_shape = (3, img_width, img_height)
else:
    input_shape = (img_width, img_height, 3)
​
model = Sequential()
model.add(Conv2D(32, (3, 3), input_shape=input_shape))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
​
model.add(Conv2D(32, (3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
​
model.add(Conv2D(64, (3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
​
model.add(Flatten())
model.add(Dense(64))
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(1))
model.add(Activation('sigmoid'))
​
model.compile(loss='binary_crossentropy',
              optimizer='rmsprop',
              metrics=['accuracy'])
​
#this is the augmentation configuration we will use for training
train_datagen = ImageDataGenerator(
    rescale=1. / 255,
    shear_range=0.2,
    zoom_range=0.2,
    horizontal_flip=True)
​
#this is the augmentation configuration we will use for testing:
#only rescaling
test_datagen = ImageDataGenerator(rescale=1. / 255)
​
train_generator = train_datagen.flow_from_directory(
    training_path,
    target_size=(img_width, img_height),
    batch_size=batch_size,
    class_mode='binary')
​
validation_generator = test_datagen.flow_from_directory(
    validation_path,
    target_size=(img_width, img_height),
    batch_size=batch_size,
    class_mode='binary')
​
model.fit_generator(
    train_generator,
    steps_per_epoch=nb_train_samples // batch_size,
    epochs=epochs,
    validation_data=validation_generator,
    validation_steps=nb_validation_samples // batch_size)
​
model.save_weights('model.h5')

model.save('model_full.h5')

What I have also tried is using the softmax activation function instead of the sigmoid one and using 'categorical_crossentropy' instead of the binary_crossentropy loss function in the training of the model.

Any help would be appreciated!
